### ISSUE TITLE
Context.IsSet returns false even if StringSlice flag is set

### ISSUE THREAD

**Issue by disktnk** [codegangsta/cli#314]
On my environment (mac OS X 10.10.5), `Context.IsSet` does not work when using `StringSliceFlag` .

```go
// `cmd` is a "sub_command" object
cmd.Flags = []cli.Flag{
	cli.StringFlag{
		Name:   "string_arg, s",
		Value:  "",
		Usage:  "snip",
	},
	cli.StringSliceFlag{
		Name:  "string_slice_arg, a",
		Value: &cli.StringSlice{},
		Usage: "snip",
	},
}

// ...snip

fmt.Println(c.IsSet("string_arg"))
fmt.Println(c.IsSet("string_slice_arg"))
```

```bash
./go_binary sub_command -s str1 -a str2
true
false
```

**Comment by meatballhat** [comment]
Confirmed!  Oddly, I'm seeing different behavior when calling `IsSet` with "short" vs "long" flag names (see [gist](https://gist.github.com/meatballhat/62a6c777148fae6bad9f0b7a3f4180da)).  Still investigating...

**Comment by meatballhat** [comment]
😑 https://github.com/codegangsta/cli/blob/222ec6fb7eb8a3da24cfa4017694ba45e3f1f516/context.go#L363-L369

**Comment by jszwedko** [comment]
Hmmm, interesting, seems to have been introduced in https://github.com/codegangsta/cli/pull/73

**Comment by meatballhat** [comment]
Yeppers.  My initial attempt to solve this is made more awkward by the slice types being aliases rather than composites.  It might be easiest to address this after the merge of https://github.com/codegangsta/cli/pull/350.

**Comment by meatballhat** [comment]
I'm going to see if I can tack this bugfix onto #350 yep yep.

**Comment by jszwedko** [comment]
Fixed by #392 (will be released is v2, but you can use that branch for now if you need the functionality sooner!).
